src/components/grid/grid-container/grid-container.d.ts

```diff
@@ -0,0 +1,32 @@
+import { GridItemProps } from "../grid-item/grid-item";
+
+export interface GridContainerProps {
+  /** Defines the Components to be rendered within the GridContainer. Requires GridItemProps */
+  children: Array<React.ReactElement<GridItemProps>> | React.ReactElement<GridItemProps>;
+  /** Any valid CSS value or a number to be multiplied by base spacing unit (8). Overrides default padding */
+  p?: string | number;
```

Comment:
This could just extend `PaddingProps` instead of having all these defined

Reply:
this would be valid if the original GridContainer would implement the `styled-system` as by extending that we'll allow use of props not available in the js propTypes, e.g.: `paddingLeft` or `paddingX`